src/riak_cs_s3_auth.erl

```diff
@@ -43,13 +44,21 @@
 
 -spec identify(RD::term(), #context{}) ->
                       {string() | undefined,
-                       string() | {v4, v4_attrs()} | undefined}.
+                       string() | {v4, v4_attrs()} | undefined} |
+                      {failed, Reason::atom()}.
 identify(RD,_Ctx) ->
     case wrq:get_req_header("authorization", RD) of
         undefined ->
             identify_by_query_string(RD);
         AuthHeader ->
-            parse_auth_header(AuthHeader)
+            %% Query string auth does not require date/x-amz-date header
+            Date = riak_cs_wm_utils:extract_date(RD),
+            case riak_cs_wm_utils:check_timeskew(Date) of
```

Comment:
Just a proposal: How about move this check logic from `identify` to `authenticate`?
Main reason is skewed time errors should be counted for access stats.
But it will be minor parts, so feel free to reject this proposal if it complicate the source code.